dali/operators/generic/reduce/sum.h

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_GENERIC_REDUCE_SUM_H__
+#define DALI_OPERATORS_GENERIC_REDUCE_SUM_H__
+
+#include "include/dali/core/static_map.h"
+#include "dali/operators/generic/reduce/reduce.h"
+
+#define SUM_TYPES_MAP ( \
```

Comment:
Please add a comment what is the format and what this map is for.

Reply:
Done. Added comment here and more elaborate doc style comments for TYPE_MAP with its implementation.